qt/aqt/webview.py

```diff
@@ -67,15 +67,24 @@ def cmd(self, str):
         script.setRunsOnSubFrames(False)
         self.profile().scripts().insert(script)
 
-    def javaScriptConsoleMessage(self, lvl, msg, line, srcID):
+    def javaScriptConsoleMessage(self, level, msg, line, srcID):
         # not translated because console usually not visible,
         # and may only accept ascii text
-        buf = "JS error on line %(a)d: %(b)s" % dict(a=line, b=msg + "\n")
+        srcID = srcID.replace(AnkiWebView.webBundlePath("/"), "")
+        if level == QWebEnginePage.InfoMessageLevel:
+            level = "info"
+        elif level == QWebEnginePage.WarningMessageLevel:
+            level = "warning"
+        elif level == QWebEnginePage.ErrorMessageLevel:
+            level = "error"
+        buf = "JS %(t)s /%(f)s:%(a)d %(b)s" % dict(
+            t=level, a=line, f=srcID, b=msg + "\n"
+        )
         # ensure we don't try to write characters the terminal can't handle
-        buf = buf.encode(sys.stdout.encoding, "backslashreplace").decode(
-            sys.stdout.encoding
+        buf = buf.encode(sys.stderr.encoding, "backslashreplace").decode(
```

Comment:
I believe switching to stderr will cause an error popup. There are shared decks floating around with bad Javascript, and changing to this behaviour is likely to cause an increase in posts to the support site

Reply:
I do not know anki deep enough to know this will cause problems, so I am reverting it back. I had set it to stderr because by convention, we should send to stdout only program results, computations, etc. While stderr is for everything else.